rotkehlchen/globaldb/handler.py

```diff
@@ -1390,6 +1380,35 @@ def get_binance_pairs(self, location: Location) -> List['BinancePair']:
                 log.debug(f'Failed to deserialize binance pair {pair}. {str(e)}')
         return pairs
 
+    def get_user_added_assets(
+        self,
+        user_db: 'DBHandler',
+        only_owned: bool = False,
+    ) -> Set[str]:
+        """
+        Create a list of the assets indentifiers added by the user. If only_owned
+        the assets added by the user and at some point he had are returned.
+        May raise:
+        - sqlite3.Error if the user_db couldn't be correctly attached
+        """
+        connection = self._conn
+        cursor = connection.cursor()
+        root_dir = Path(__file__).resolve().parent.parent
+        builtin_database = root_dir / 'data' / 'global.db'
+        # Update the list of owned assets
+        user_db.update_owned_assets_in_globaldb()
+        if only_owned:
+            query = cursor.execute('SELECT asset_id from user_owned_assets;')
+        else:
+            query = cursor.execute('SELECT identifier from assets;')
+        user_ids = {tup[0] for tup in query.fetchall()}
```

Comment:
fetchall() iterates the entire query result. So you have a double looping here. Remove it.
```suggestion
        user_ids = {tup[0] for tup in query}
```